### PR TITLE
Add a recipe for teensy_loader_cli

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vend other packages
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/teensy_loader_cli/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+build:
+  number: 0
+  # For now, start with linux builds ony
+  skip: True
+  script:
+    # http://www.linuxfromscratch.org/blfs/view/6.3/general/libusb.html
+    - make OS=LINUX    # [linux]
+    - make OS=MACOSX   # [osx]
+    - cp teensy_loader_cli  {{ PREFIX }}/bin/teensy_loader_cli  # [unix]
+    - make OS=WINDOWS  # [windows]
```
Comment (marcelotrevisani) [inline recipes/teensy_loader_cli/meta.yaml]:
```suggestion
    - make OS=WINDOWS  # [win]
```

File: recipes/teensy_loader_cli/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+source:
+  git_url: https://github.com/PaulStoffregen/teensy_loader_cli.git
+  # Build off a recent master since 2019/08/28
+  git_rev: eb61aa1a482c2ad98916229e02bb8caae2464f48
+
+build:
+  number: 0
+  # For now, start with linux builds ony
+  skip: True
```
Comment (marcelotrevisani) [inline recipes/teensy_loader_cli/meta.yaml]: This is skipping everything
Should it be ?
```suggestion
  skip: True  # [win]
```

**Comment by hmaarrfk** [comment]
needs a patch for windows to build with our compiler,

File: recipes/teensy_loader_cli/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - hid_location_windows.patch  # [win]
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}        # [not win]
+    - {{ compiler('m2w64_c') }}  # [win]
+    # - m2w64-toolchain   # [win]
```
Comment (hmaarrfk) [inline recipes/teensy_loader_cli/meta.yaml]:
```suggestion
```

**Comment by hmaarrfk** [comment]
@marcelotrevisani this is ready now for review if you were still interested.

**Pull request merged**